includes/options/views/class-amp-setup-wizard-submenu-page.php

```diff
@@ -0,0 +1,223 @@
+	 * @var string
+	 */
+	const JS_HANDLE = 'amp-setup';
+
+	/**
+	 * HTML ID for the app root element.
+	 *
+	 * @var string
+	 */
+	const APP_ROOT_ID = 'amp-setup';
+
+	/**
+	 * The minimum core WP version whose core assets are compatibile with the setup feature.
+	 *
+	 * @var float
+	 */
+	const TARGET_WP_VERSION = 5.4;
+
+	/**
+	 * Sets up hooks.
+	 */
+	public function init() {
+		add_action( 'admin_enqueue_scripts', [ $this, 'override_scripts' ] );
+		add_action( 'admin_enqueue_scripts', [ $this, 'enqueue_assets' ] );
+	}
+
+	/**
+	 * Renders the setup screen markup.
+	 */
+	public function render() {
+		?>
+			<div id="<?php echo esc_attr( static::APP_ROOT_ID ); ?>"></div>
+		<?php
+	}
+
+	/**
+	 * Provides the setup screen handle.
+	 *
+	 * @return string
+	 */
+	public function screen_handle() {
+		return sprintf( 'amp_page_%s', AMP_Setup_Wizard_Submenu::SCREEN_ID );
+	}
+
+	/**
+	 * Registers single script.
+	 *
+	 * @param string  $handle  Script handle.
+	 * @param boolean $enqueue Whether to enqueue the asset immediately. Default false.
+	 */
+	public function add_setup_script( $handle, $enqueue = false ) {
+		$asset = $this->get_asset( $handle, sprintf( '%s/assets/js/%s.asset.php', AMP__DIR__, $handle ) );
+
+		wp_register_script(
+			$handle,
+			amp_get_asset_url( sprintf( 'js/%s.js', $handle ) ),
+			$asset['dependencies'],
+			$asset['version'],
+			true
+		);
+
+		if ( $enqueue ) {
+			wp_enqueue_script( $handle );
+		}
+	}
+
+	/**
+	 * Returns an asset's dependencies and version.
+	 *
+	 * @param string $handle Asset handle.
+	 * @param string $path   Path to file containing asset details.
+	 *
+	 * @return array Associative array containing the asset's dependencies and version.
+	 */
+	public function get_asset( $handle, $path = null ) {
+		/**
+		 * Filters AMP asset details.
+		 *
+		 * @param null|array $asset  Null or, to override, an array containing the asset's dependencies and version string.
+		 * @param string     $handle The asset handle.
+		 */
+		$asset = apply_filters( 'amp_setup_asset', null, $handle );
+
+		if ( is_null( $asset ) ) {
+			$asset_file = $path;
+			$asset      = require $asset_file;
+		}
+
+		return $asset;
+	}
+
+	/**
+	 * Enqueues setup assets.
+	 *
+	 * @param string $hook_suffix The current admin page.
+	 */
+	public function enqueue_assets( $hook_suffix ) {
+		if ( $this->screen_handle() !== $hook_suffix ) {
+			return;
+		}
+
+		$this->add_setup_script( static::JS_HANDLE, true );
+
+		wp_localize_script(
+			static::JS_HANDLE,
+			'ampSetup',
+			[
+				'APP_ROOT_ID' => static::APP_ROOT_ID,
+			]
+		);
+	}
+
+	/**
+	 * Overrides core assets with required versions if needed.
+	 *
+	 * @param string $hook_suffix The current admin page.
+	 */
+	public function override_scripts( $hook_suffix ) {
+		global $wp_version;
+
+		if ( $this->screen_handle() !== $hook_suffix ) {
+			return;
+		}
+
+		if ( floatval( $wp_version ) >= static::TARGET_WP_VERSION && wp_script_is( 'react', 'registered' ) ) {
```

Comment:
Avoiding late static binding as not needed when class is made `final`:
```suggestion
		if ( floatval( $wp_version ) >= self::TARGET_WP_VERSION && wp_script_is( 'react', 'registered' ) ) {
```